### DISCUSSION TITLE
Standard Umbraco Publish Dates on Element Types - Particularly for Block List and Grid

### DISCUSSION THREAD

**Discussion by deanleigh** [umbraco/Umbraco-CMS#13516] [Features and ideas]
TLDR: Please add the same publish date functionality that is used for document types to element types.

We I've found since using Umbraco that we have to add our own publish and archive dates two items all the way from nested content, doc type grid editor, block list items and now it would appear for block grid items.

Whilst this works to degree we do not have a way of passing the created date of the element type to the published date in the same way we can with document types thereby creating a poor user experience.

We feel that if a different type of publish mechanism is used with element types this would not be an ideal solution.
It would surely be best practise creating something new that all types of publish and archive functionality should work the same way.

I hope I have been clear in my explanation, please let me know if I have not.

**Comment by lassefredslund** [comment]
Hi @deanleigh.

Thanks for reaching out.

I'm not sure I fully understand what you want to achieve here. 
If possible, could you then please elaborate briefly on the use case you are trying to solve?

**Comment by deanleigh** [comment]
Thanks @lassefredslund,
the use case is to give the editor the abilty, out of the box to add publish and archive dates to blocks in the same way as pages.
This is funtionality common in almost every CMS big or small.
Currently most developers are having to add this functionality manually in block settings using custon view e.g. 
[https://archive.24days.in/umbraco-cms/2021/advanced-blocklist-editor/](https://archive.24days.in/umbraco-cms/2021/advanced-blocklist-editor/)

If this implemented, ideally for a good user experience it would use the same UI component for a Block as publishing a Page.

Happy to explain further or give a demo in one of our builds

**Comment by lassefredslund** [comment]
OK. Thanks. That makes sense. I think/hope that what you want is covered in the "Reusable Block Content" which we have on the roadmap for V13 (See short description here: https://umbraco.com/products/roadmap/). The design/discussions of this are still very early days, but I think we'll have to introduce a publish option directly on the reusable blocks just like you ask for.

I'll reach out to you for a short demo of your custom implementation.
